core/app/assets/js/ui/modal/modal.js

```diff
@@ -78,20 +80,40 @@ export const Modal = ( props ) => {
 			const node = modalRef.current,
 				closeNode = closeRef.current,
 				isInCloseNode = closeNode && closeNode.contains( e.target );
-
 			// Ignore if click is inside the modal
 			if ( node && node.contains( e.target ) && ! isInCloseNode ) {
 				return;
 			}
 
 			props.hideModal();
+
+			// TODO: distinguish event type on modal close: overlay or close button
+			// if ( 'kit-library' === props.referrer && props.onClose ) {
+			// 	if ( e.target.classList.contains( 'eps-modal__overlay' ) ) {
+			// 		props.onClose( 'overlay' );
+			// 		console.log( 'close overlay' )
+			// 	} else {
+			// 		console.log( 'close button' )
+			// 		props.onClose( 'button' );
+			// 	}
+			// }
+
+			// TODO: distinguish event type on modal close: overlay or close button
+			if ( props.onClose ) {
+				props.onClose( e );
+			}
 		};
 
 	useEffect( () => {
 		if ( props.show ) {
 			document.addEventListener( 'mousedown', closeModal, false );
 		}
 
+		// Kit library event tracking
+		if ( props.onOpen && props.show ) {
+			props.onOpen();
+		}
```

Comment:
you can move it to the condition above and just do this:
```JS
props.onOpen.?()
```

or add a default prop and remove the `?`